test/wtr/app/plugins/preview/preview.test.js

```diff
@@ -0,0 +1,203 @@
+import { EventBus } from '../../../../../src/extension/app/utils/event-bus.js';
+import { EVENTS, MODALS } from '../../../../../src/extension/app/constants.js';
+
+// @ts-ignore
+window.chrome = chromeMock;
+
+describe('Preview plugin', () => {
+  let sidekick;
+  beforeEach(async () => {
+    mockFetchEnglishMessagesSuccess();
+  });
+
+  afterEach(() => {
+    document.body.removeChild(sidekick);
+    fetchMock.reset();
+    restoreEnvironment(document);
+  });
+
+  describe('switching between environments', () => {
+    it('previewing from sharepoint editor - docx', async () => {
+      mockSharepointEditorDocFetchStatusSuccess();
+      mockFetchConfigJSONNotFound();
+      mockEditorAdminEnvironment(document, 'editor');
+      const updatePreviewSpy = sinon.stub(appStore, 'updatePreview').resolves();
+      const tipToast = sinon.stub(appStore, 'showToast').returns();
+
+      sidekick = new AEMSidekick(defaultSidekickConfig);
+      document.body.appendChild(sidekick);
+
+      await waitUntil(() => recursiveQuery(sidekick, 'action-bar-picker'));
+
+      const previewPlugin = recursiveQuery(sidekick, '.edit-preview');
+      expect(previewPlugin.textContent.trim()).to.equal('Preview');
+
+      previewPlugin.dispatchEvent(new Event('click'));
+
+      expect(updatePreviewSpy.calledOnce).to.be.true;
+      expect(tipToast.calledOnce).to.be.true;
+
+      updatePreviewSpy.restore();
+      tipToast.restore();
+    });
+
+    it('previewing from sharepoint editor - sheet', async () => {
+      mockSharepointEditorSheetFetchStatusSuccess();
+      mockFetchConfigJSONNotFound();
+      mockEditorAdminEnvironment(document, 'editor', HelixMockContentType.SHEET);
+      const updatePreviewSpy = sinon.stub(appStore, 'updatePreview').resolves();
+      const reloadStub = sinon.stub(appStore, 'reloadPage').returns();
+
+      sidekick = new AEMSidekick(defaultSidekickConfig);
+      document.body.appendChild(sidekick);
+
+      await waitUntil(() => recursiveQuery(sidekick, 'action-bar-picker'));
+
+      // Click the preview the plugin
+      const previewPlugin = recursiveQuery(sidekick, '.edit-preview');
+      previewPlugin.dispatchEvent(new Event('click'));
+
+      expect(reloadStub.calledOnce).to.be.true;
+
+      // Simulate a reload
+      document.body.removeChild(sidekick);
+
+      // Make sure the hlx-sk-preview flag is set
+      expect(window.sessionStorage.getItem('hlx-sk-preview')).to.not.be.null;
+
+      // Reset the environment
+      sidekick = new AEMSidekick(defaultSidekickConfig);
+      document.body.appendChild(sidekick);
+
+      await waitUntil(() => recursiveQuery(sidekick, 'action-bar-picker'));
+
+      expect(updatePreviewSpy.calledOnce).to.be.true;
+
+      // Make sure the hlx-sk-preview flag is unset
+      expect(window.sessionStorage.getItem('hlx-sk-preview')).to.be.null;
+
+      updatePreviewSpy.restore();
+      reloadStub.restore();
+    });
```

Comment:
Should we also test if the plugin tried to open a new window with the expected preview URL?

Reply:
Between[ this test](https://github.com/adobe/aem-sidekick/blob/3bdc760a20ad98d31aba30f31801af4c98d69e51/test/wtr/app/store/app.test.js#L562) and the ones that still need to be written for `switchEnv` in `appStore` I think we will have tested that scenario (indirectly of course).

I believe there is an issue with opening a new window with web-test-runner, it seems to stop the tests since that new tab gets focus. So I think no matter what best we can go is stub `window.location` and check that the stub was called with the correct arguments.

wdyt?